x-pack/test/lists_api_integration/security_and_spaces/tests/create_lists_index.ts

```diff
@@ -46,14 +46,15 @@ export default ({ getService }: FtrProviderContext) => {
       expect(body).to.eql({ list_index: true, list_item_index: true });
     });
 
-    it('should update lists indices if old legacy templates exists', async () => {
+    it('should migrate lists indices to data streams and remove old legacy templates', async () => {
```

Comment:
Could you add test suites ensuring that the API operations, e.g. update list item, work on legacy list data? I was experimenting with updateByQuery on a migrated lists index and encountered this exception
```
{
      "index": ".items-default-000001",
      "id": "QxUPvYkByw4H-DpCsc2_",
      "cause": {
        "type": "document_parsing_exception",
        "reason": "[1:239] failed to parse: data stream timestamp field [@timestamp] is missing",
        "caused_by": {
          "type": "illegal_argument_exception",
          "reason": "data stream timestamp field [@timestamp] is missing"
        }
      },
      "status": 400
    }
```
when my updateByQuery script did not add a timestamp field.

Reply:
good catch, I have updated relevant APIs and added test coverage for it.
I also introduced migration to the rest of methods, this way we won't need to support 2 different scenarios: when to add missing `@timestamp` or not to add, if document belongs to migrated legacy index